ToNote/Models/ToDo.cs

```diff
@@ -35,5 +45,32 @@ public int Index
                 }
             }
         }
+
+        private DateTime? _SelectedDate;
+
+        public DateTime? SelectedDate
+        {
+            get => _SelectedDate;
+            set
+            {
+                if (_SelectedDate != value)
+                {
+                    _SelectedDate = value;
+
+                    RaisePropertyChanged(nameof(SelectedDate));
+                    RaisePropertyChanged(nameof(IsDatePast));
+                }
+                
+            }
+        }
+
+        private DateTime? _LastDone;
+
+        public DateTime? LastDone
+        {
+            get => _LastDone;
+        }
+
+        public bool IsDatePast => _SelectedDate != null ? DateTime.Compare((DateTime)_SelectedDate, DateTime.Today) > 0 ? true : false : false;
```

Comment:
Refactor spaghetti to `public bool IsDatePast => _SelectedDate.HasValue ? (DateTime.Compare(_SelectedDate.Value, DateTime.Today) >= 0) : false;`. Notice that I changed > to >=, I don't think a task that is due today should be marked as overdue.